code/controllers/subsystem/timer.dm

```diff
@@ -42,38 +52,44 @@ var/datum/subsystem/timer/SStimer
 	var/static/nextid = 1
 
 /datum/timedevent/New()
-	id = nextid
-	nextid++
+	id = nextid++
 
 /datum/timedevent/Destroy()
 	SStimer.processing -= src
 	SStimer.hashes -= hash
 	return QDEL_HINT_IWILLGC
 
 /proc/addtimer(thingToCall, procToCall, wait, unique = FALSE, ...)
-	if (!SStimer) //can't run timers before the mc has been created
-		return
-	if (!thingToCall || !procToCall || wait <= 0)
+	if (!thingToCall || !procToCall)
 		return
 	if (!SStimer.can_fire)
 		SStimer.can_fire = 1
-		SStimer.next_fire = world.time + SStimer.wait
 
 	var/datum/timedevent/event = new()
 	event.thingToCall = thingToCall
 	event.procToCall = procToCall
 	event.timeToRun = world.time + wait
+	var/hashlist = args.Copy()
+
+	hashlist[1] = "[thingToCall](\ref[thingToCall])"
 	event.hash = jointext(args, null)
 	if(args.len > 4)
 		event.argList = args.Copy(5)
 
 	// Check for dupes if unique = 1.
 	if(unique)
-		if(event.hash in SStimer.hashes)
-			return
+		var/datum/timedevent/hash_event = SStimer.hashes[event.hash]
+		if(hash_event)
+			return hash_event.id
+	SStimer.hashes[event.hash] = event
+	if (wait <= 0)
+		spawn
```

Comment:
I'm suprised this is valid syntax and also, what on earth is it doing here? wouldn't it be better to setbackground or something?